### PR TITLE
feat(clever): add clever as a login and signup option

### PR DESCRIPTION
**Pull request by stephenliang**
This change exposes the existing Clever login omniauth callbacks to the sign in page. Previously, these weren't exposed due to the potential confusion caused by the district picker on Clever however this login pattern might be well known to  schools as other companies use the same mechanism for signin with Clever.

## Links

- [jira ticket](https://codedotorg.atlassian.net/browse/P20-957)

## Testing story

Tested with student and teacher login

https://github.com/code-dot-org/code-dot-org/assets/538214/a1425d16-3173-477e-8468-749535bdf596

https://github.com/code-dot-org/code-dot-org/assets/538214/22483938-1bcc-43a7-b18a-6958914125d6

## Statsig Instrumentation

To test the Statsig instrumentation, I ran through sign up and sign in procedures for all providers, the output is attached:

```
I, [2024-06-03T15:01:59.961661 #177122]  INFO -- omniauth: (clever) Request phase initiated.
Logging Event: {"user_id":"","custom_ids":{"user_type":""},"event_name":"clever-begin-auth","event_value":"clever-begin-auth","metadata":{}}
I, [2024-06-03T15:02:00.285819 #177122]  INFO -- omniauth: (clever) Callback phase initiated.
Logging Event: {"user_id":"","custom_ids":{"user_type":""},"event_name":"clever-callback","event_value":"clever-callback","metadata":{"study":"account-sign-up-v5","study_group":"not-in-study"}}
WARNING: MYSQL_OPT_RECONNECT is deprecated and will be removed in a future version.
I, [2024-06-03T15:02:18.415442 #177122]  INFO -- omniauth: (facebook) Request phase initiated.
Logging Event: {"user_id":"","custom_ids":{"user_type":""},"event_name":"facebook-begin-auth","event_value":"facebook-begin-auth","metadata":{}}
I, [2024-06-03T15:02:23.680874 #177122]  INFO -- omniauth: (google_oauth2) Setup endpoint detected, running now.
I, [2024-06-03T15:02:23.681514 #177122]  INFO -- omniauth: (google_oauth2) Request phase initiated.
Logging Event: {"user_id":"","custom_ids":{"user_type":""},"event_name":"google_oauth2-begin-auth","event_value":"google_oauth2-begin-auth","metadata":{}}
I, [2024-06-03T15:02:27.020300 #177122]  INFO -- omniauth: (microsoft_v2_auth) Request phase initiated.
Logging Event: {"user_id":"","custom_ids":{"user_type":""},"event_name":"microsoft_v2_auth-begin-auth","event_value":"microsoft_v2_auth-begin-auth","metadata":{}}
Logging Event: {"user_id":"","custom_ids":{"user_type":""},"event_name":"load-sign-up-page","event_value":"load-sign-up-page","metadata":{"study":"account-sign-up-v5","study_group":"control-v4"}}
I, [2024-06-03T15:02:43.621483 #177122]  INFO -- omniauth: (clever) Request phase initiated.
Logging Event: {"user_id":"","custom_ids":{"user_type":""},"event_name":"clever-begin-auth","event_value":"clever-begin-auth","metadata":{}}
I, [2024-06-03T15:02:43.835093 #177122]  INFO -- omniauth: (clever) Callback phase initiated.
Logging Event: {"user_id":"","custom_ids":{"user_type":""},"event_name":"clever-signup-callback","event_value":"clever-signup-callback","metadata":{"study":"account-sign-up-v5","study_group":"control-v4"}}
Logging Event: {"user_id":"","custom_ids":{"user_type":""},"event_name":"clever-sign-in","event_value":"clever-sign-in","metadata":{"study":"account-sign-up-v5","study_group":"control-v4"}}
I, [2024-06-03T15:03:36.480637 #177122]  INFO -- omniauth: (clever) Request phase initiated.
Logging Event: {"user_id":"","custom_ids":{"user_type":""},"event_name":"clever-begin-auth","event_value":"clever-begin-auth","metadata":{}}
I, [2024-06-03T15:03:49.468307 #177122]  INFO -- omniauth: (clever) Callback phase initiated.
Logging Event: {"user_id":"","custom_ids":{"user_type":""},"event_name":"clever-signup-callback","event_value":"clever-signup-callback","metadata":{"study":"account-sign-up-v5","study_group":"not-in-study"}}
Logging Event: {"user_id":"","custom_ids":{"user_type":""},"event_name":"clever-sign-up-success","event_value":"clever-sign-up-success","metadata":{"study":"account-sign-up-v5","study_group":"not-in-study"}}
```

## Privacy

No net new feature, we already have Clever as a login option but was not exposed as a public facing signin option.

## Security

No net new feature, we already have Clever as a login option but was not exposed as a public facing signin option.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by nicklathe — APPROVED** [review]
This looks good from my perspective

File: dashboard/lib/sign_up_tracking.rb [hunk 1]
```diff
@@ -71,6 +79,14 @@ def self.log_begin_sign_up(user, session)
       }.to_json
     }
     FirehoseClient.instance.put_record(:analysis, tracking_data)
+
+    Metrics::Events.log_event(
+      event_name: "#{provider}-begin-sign-up-#{result}",
```
Comment (nicklathe) [inline dashboard/lib/sign_up_tracking.rb]: Potentially dumb question (might be a brain glitch), where is `provider` defined in the scope of this method? I see it's a parameter in the methods below, but not sure it's defined here?

Comment (stephenliang) [reply dashboard/lib/sign_up_tracking.rb]: You're right, I somehow missed this and removed it. Thank you!

**Comment by nicklathe** [comment]
LGTM :shipit:

**Pull request merged**